internal/util/taints/taints_test.go

```diff
@@ -55,6 +55,16 @@ func TestRemoveNodeTaint(t *testing.T) {
 			wantTaints:   []corev1.Taint{taint2},
 			wantModified: false,
 		},
+		{
+			name: "drop last taint should return true",
```

Comment:
Just curious: What case does this cover that other tests are not covering?

Reply:
While I was writing the other unit test I saw that when only the node taint that gets dropped is the only one there the node has nil Taints afterwards (https://github.com/kubernetes-sigs/cluster-api/pull/8462/files#diff-a86dcfba31aabdefe7ddd3c9e7d33901fcac8e69ed33523a8ed0ef59aa07ad74R320) so I wanted to verify if we were setting them to nil or if something else was happening in the patching. I think it's the latter but I just left the unit test there so it's explicit.